src/wazuh_db/wdb_global.c

```diff
@@ -110,3 +134,68 @@ int wdb_global_set_agent_label(wdb_t *wdb, int id, char* key, char* value) {
         return OS_INVALID;
     }
 }
+
+int wdb_global_update_unsynced_agents(wdb_t *wdb,cJSON * json_agent){
+    sqlite3_stmt *stmt = NULL;
+    int n = 0;
+    int agent_id = 0;
+    cJSON *json_field = NULL;
+
+    if (!wdb->transaction && wdb_begin2(wdb) < 0) {
+        mdebug1("Cannot begin transaction");
+        return OS_INVALID;
+    }
+
+    if (wdb_stmt_cache(wdb, WDB_STMT_GLOBAL_UPDATE_UNSYNCED_AGENTS) < 0) {
+        mdebug1("Cannot cache statement");
+        return OS_INVALID;
+    }
+
+    stmt = wdb->stmt[WDB_STMT_GLOBAL_UPDATE_UNSYNCED_AGENTS];
+
+     for (n = 0 ; global_db_agent_fields[n] ; n++){
+        // Every column name of Global DB is stored in global_db_agent_fields 
+        json_field = cJSON_GetObjectItemCaseSensitive(json_agent, global_db_agent_fields[n]);
+        if (cJSON_IsNumber(json_field)){
+            if (sqlite3_bind_int(stmt, n + 1 , json_field->valueint) != SQLITE_OK) {
+                merror("DB(%s) sqlite3_bind_int(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+                return OS_INVALID;
+            }
+       
+        } else if (cJSON_IsString(json_field) && json_field->valuestring != NULL) {
+            if (sqlite3_bind_text(stmt, n + 1 , json_field->valuestring, -1, NULL) != SQLITE_OK) {
+                merror("DB(%s) sqlite3_bind_text(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+                return OS_INVALID;
+            }
+        
+        } else{
+            if (sqlite3_bind_null(stmt, n+1) != SQLITE_OK) {
```

Comment:
Just by curious... Is this to handle the case where for an agent, one of the fields does not contain information?

Reply:
This is unnecessary, removed!